src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -172,9 +175,9 @@ private Optional<Optional<String>> getCompanyNumberInTransaction(HttpServletRequ
             final var transaction = transactionService.getTransaction(transactionId, passthroughHeader, reqId);
             logMap.put(COMPANY_NUMBER_KEY, transaction.getCompanyNumber());
             ApiLogger.debugContext(reqId, "Transaction successfully retrieved " + transactionId, logMap);
-            return Optional.of(Optional.ofNullable(transaction.getCompanyNumber()));
+            return Optional.of(transaction.getCompanyNumber());
```

Comment:
Can tranaction.getCompanyNumber() not return null? 
In which case Optiona.of(null) would throw

Reply:
Yeah that's Martin's approach to let it throw and deal with it as a 500 status

Reply:
Yes, exactly.